pkg/gitprovider/gitness.go

```diff
@@ -281,3 +297,96 @@ func (g *GitnessGitProvider) GetDefaultBranch(staticContext *StaticGitContext) (
 	client := g.getApiClient()
 	return client.GetDefaultBranch(staticContext.Url)
 }
+
+func (g *GitnessGitProvider) RegisterPrebuildWebhook(repo *GitRepository, endpointUrl string) (string, error) {
+	client := g.getApiClient()
+	webhook, err := client.CreateWebhook(repo.Id, repo.Owner, gitnessclient.Webhook{
+		Triggers:    []string{"branch_updated"},
+		Url:         endpointUrl,
+		Identifier:  "daytona-webhook_" + repo.Id,
+		DisplayName: "Daytona Webhook",
+		Enabled:     true,
+	})
+	if err != nil {
+		return "", err
+	}
+	return webhook.Uid, nil
+}
+
+func (g *GitnessGitProvider) GetPrebuildWebhook(repo *GitRepository, endpointUrl string) (*string, error) {
+	client := g.getApiClient()
+	webhooks, err := client.GetAllWebhooks(repo.Id, repo.Owner)
+	if err != nil {
+		return nil, err
+	}
+	for _, webhook := range webhooks {
+		if webhook.Url == endpointUrl {
+			return &webhook.Uid, nil
+		}
+	}
+	return nil, nil
+}
+
+func (g *GitnessGitProvider) UnregisterPrebuildWebhook(repo *GitRepository, id string) error {
+	client := g.getApiClient()
+	return client.DeleteWebhook(repo.Id, repo.Owner, id)
+}
+
+func (g *GitnessGitProvider) GetCommitsRange(repo *GitRepository, initialSha string, currentSha string) (int, error) {
+	client := g.getApiClient()
+	commits, err := client.GetCommits(repo.Owner, repo.Name, &repo.Branch)
+	if err != nil {
+		return 0, err
+	}
+	initialShaIndex := -1
+	currentShaIndex := -1
+	for i := 0; i < len(*commits); i++ {
+		if currentSha == (*commits)[i].Sha {
+			currentShaIndex = i
+		}
+		if initialSha == (*commits)[i].Sha {
+			initialShaIndex = i
+		}
+		if initialShaIndex != -1 && currentShaIndex != -1 {
+			break
+		}
+	}
+
+	if initialShaIndex == -1 || currentShaIndex == -1 {
+		return 0, fmt.Errorf("Sha Not found in commits")
+	}
+
+	commitLength := int(math.Abs(float64(initialShaIndex - currentShaIndex)))
+
+	return commitLength, nil
+}
+
+func (g *GitnessGitProvider) ParseEventData(request *http.Request) (*GitEventData, error) {
+	payload, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+	var webhookEvent gitnessclient.WebhookEventData
+	err = json.Unmarshal(payload, &webhookEvent)
+	if err != nil {
+		return nil, err
+	}
+
+	if webhookEvent.Trigger != "branch_updated" {
+		return nil, nil
+	}
+
+	gitEventData := &GitEventData{
+		Url:    webhookEvent.Repo.GitURL,
```

Comment:
Here - check GitHub implementation

Reply:
Getting ".git.git" now. Please attach a video of this working after removing the extra + ".git" 

![image](https://github.com/user-attachments/assets/72f6a956-85cd-4e0d-acda-c132e06b171d)